src/ui/image_upload.rs

```diff
@@ -44,91 +44,92 @@ impl<'a> Widget for ImageUpload<'a> {
                     Stroke::new(self.rounding.min(self.size * 0.5), ui.visuals().window_fill),
                 );
 
-                if self.editable || self.removable {
-                    if ui.rect_contains_pointer(rect) {
-                        ui.put(rect, |ui: &mut Ui| {
-                            ui.painter().rect_filled(
-                                rect,
-                                self.rounding,
-                                Color32::from_rgba_premultiplied(20, 20, 20, 180),
+                let editable = self.editable;
+                let removable = self.removable;
+                let contains_pointer = ui.rect_contains_pointer(rect);
+
+                if (editable || removable) && contains_pointer {
+                    ui.put(rect, |ui: &mut Ui| {
+                        ui.painter().rect_filled(
+                            rect,
+                            self.rounding,
+                            Color32::from_rgba_premultiplied(20, 20, 20, 180),
+                        );
+
+                        let mut content_size = Vec2::default();
+                        let button_padding = ui.spacing().button_padding;
+
+                        if self.editable {
```

Comment:
Seems weird using `editable` in L51, but uses `self.editable` again here. Same L72, L82, L98, L114.